src/ui/public/inspector/build_tabular_inspector_data.js

```diff
@@ -17,41 +17,86 @@
  * under the License.
  */
 
+import { set } from 'lodash';
 import { FormattedData } from './adapters/data';
 
+const getTerms = (table, columnIndex, rowIndex) => {
+  if (rowIndex === -1) {
+    return [];
+  }
+
+  // get only rows where cell value matches current row for all the fields before columnIndex
+  const rows = table.rows.filter(row => {
+    return table.columns.every((column, i) => {
+      return row[column.id] === table.rows[rowIndex][column.id] || i >= columnIndex;
+    });
+  });
+  const terms = rows.map(row => row[table.columns[columnIndex].id]);
+
+  return [...new Set(terms.filter(term => {
+    const notOther = term !== '__other__';
+    const notMissing = term !== '__missing__';
+    return notOther && notMissing;
+  }))];
+};
+
+const createFilter = (data, columnIndex, rowIndex, cellValue) => {
+  const { aggConfig, id: columnId } = data.columns[columnIndex];
+  let filter = [];
+  const value = rowIndex > -1 ? data.rows[rowIndex][columnId] : cellValue;
+  if (value === null || value === undefined) {
+    return;
+  }
+  if (aggConfig.type.name === 'terms' && aggConfig.params.otherBucket) {
+    const terms = getTerms(data, columnIndex, rowIndex);
+    filter = aggConfig.createFilter(value, { terms });
+  } else {
+    filter = aggConfig.createFilter(value);
+  }
+  return filter;
+};
+
 /**
  * This function builds tabular data from the response and attaches it to the
  * inspector. It will only be called when the data view in the inspector is opened.
  */
 export async function buildTabularInspectorData(table, queryFilter) {
-  const columns = table.columns.map((col, index) => {
+  const rows = table.rows.map(row => {
+    return table.columns.reduce((prev, cur, colIndex) => {
+      const value = row[cur.id];
+      const fieldFormatter = cur.aggConfig.fieldFormatter('text');
+      prev[`col-${colIndex}-${cur.aggConfig.id}`] = new FormattedData(value, fieldFormatter(value));
```

Comment:
I changed the way keys were formatted in rows to be consistent with tabify:

```
{ col-0-2: 'someval', col-1-1: 'anotherval' }
```

Reply:
i think we should rather update inspector to work with tabify data
not sure how much sense it still makes for it to keep its own data format as its very close to tabify.

Reply:
Inspector still takes in tabified data, but it rewrites each row into `FormattedData`.

@timroes Maybe you have more thoughts on this? I'm not 100% clear on the intent of `FormattedData`, right now it just contains both raw and formatted cell values, but perhaps you had other designs for it in the future?

Reply:
Discussed with @timroes. He confirmed that the only intent of `FormattedData` is to work around the need to access both raw and formatted cell values.

He agreed we should try to figure out a different solution so we don't need to use `FormattedData` and can just use pure tabify data, but we decided it would make more sense for this to be a separate task, since this PR is simply addressing a bug and not introducing any "new" inspector functionality.